Chapter-2-modules-separation/Src/Fitnet.Contracts.Application/Sign/SignContractCommandHandler.cs

```diff
@@ -1,23 +1,33 @@
 namespace EvolutionaryArchitecture.Fitnet.Contracts.Application.Sign;
 
 using Common.Api.ErrorHandling;
+using Common.Core.SystemClock;
+using Common.Infrastructure.Events.EventBus;
 using Core;
 
 [UsedImplicitly]
 internal sealed class SignContractCommandHandler : IRequestHandler<SignContractCommand>
 {
     private readonly IContractsRepository _contractsRepository;
-    
-    public SignContractCommandHandler(IContractsRepository contractsRepository) => 
+    private readonly ISystemClock _systemClock;
+    private readonly IEventBus _eventBus;
+
+    public SignContractCommandHandler(IContractsRepository contractsRepository, ISystemClock systemClock, IEventBus eventBus)
+    {
         _contractsRepository = contractsRepository;
-    
+        _systemClock = systemClock;
+        _eventBus = eventBus;
+    }
+
     public async Task Handle(SignContractCommand command, CancellationToken cancellationToken)
     {
         var contract = await _contractsRepository.GetByIdAsync(command.Id, cancellationToken);
         if (contract is null)
            throw new ResourceNotFoundException(command.Id);
         
-        contract.Sign(command.SignedAt);
+        contract.Sign(command.SignedAt, _systemClock);
```

Comment:
Do not bother core (business logic with system clock that is some kind of mechanism. You do instead:

```
var expiringAt = _systemClock.Now.Add(contract.Duration);

contract.Sign(command.SignedAt, expiringAt)
```

Reply:
Calculating expiringAt outside the contract domain model violates encapsulation by exposing implementation details to the calling code. Always we caller want to use sign method needs to know how calculate contract expiration data.

```suggestion
        contract.Sign(command.SignedAt, _systemClock.Now);
```

To increase domain purity we can pass now date as parameter to method and still has logic encapsulated. What you think?

Reply:
That's fine, I like it!

Reply:
Fixed 👌